pkg/codegen/go/gen.go

```diff
@@ -165,6 +165,44 @@ func (pkg *pkgContext) tokenToType(tok string) string {
 	return strings.Replace(mod, "-provider", "", -1)
 }
 
+func (pkg *pkgContext) tokenToEnum(tok string) string {
+	// token := pkg : module : member
+	// module := path/to/module
+
+	components := strings.Split(tok, ":")
+	contract.Assert(len(components) == 3)
+	if pkg == nil {
+		panic(fmt.Errorf("pkg is nil. token %s", tok))
+	}
+	if pkg.pkg == nil {
+		panic(fmt.Errorf("pkg.pkg is nil. token %s", tok))
+	}
+
+	mod, name := pkg.tokenToPackage(tok), components[2]
+
+	modPkg, ok := pkg.packages[mod]
+	name = Title(name)
+
+	if ok {
+		newName, renamed := modPkg.renamed[name]
+		if renamed {
+			name = newName
+		} else if modPkg.duplicateTokens[tok] {
+			// If the package containing the enum's token already has a resource or type with the
+			// same name, add an `Enum` suffix.
+			name += "Enum"
```

Comment:
`modPkg.renamed[name]` edit not needed here? I have not fully grasped what that does.